chain/xlm/client/client.go

```diff
@@ -0,0 +1,407 @@
+	url := client.GetAccountUrl(string(address))
+	var response types.GetAccountResult
+	if err := client.Get(url, &response); err != nil {
+		return xc.AmountBlockchain{}, fmt.Errorf("failed to fetch account balances: %w", err)
+	}
+
+	// Asset code is omited for native currency
+	if asset == "XLM" {
+		asset = ""
+	}
+	for _, balance := range response.Balances {
+		if balance.AssetType == types.AssetTypeLiquidityPoolShares {
+			continue
+		}
+
+		if balance.AssetCode == string(asset) {
+			readableAmount, err := xc.NewAmountHumanReadableFromStr(balance.Balance)
+			if err != nil {
+				return xc.AmountBlockchain{}, fmt.Errorf("failed to read balance decimal: %w", err)
+			}
+			blockchainAmount := readableAmount.ToBlockchain(client.Asset.GetChain().GetDecimals())
+			return blockchainAmount, nil
+		}
+	}
+	return xc.AmountBlockchain{}, nil
+}
+
+func (client *Client) FetchSequenceNumber(address xc.Address) (int64, error) {
+	url := client.GetAccountUrl(string(address))
+	var response types.GetAccountResult
+	if err := client.Get(url, &response); err != nil {
+		return 0, fmt.Errorf("failed to fetch account data: %w", err)
+	}
+
+	sequence, err := strconv.ParseInt(response.Sequence, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse sequence number: %w", err)
+	}
+
+	return sequence, nil
+}
+
+func (client *Client) FetchBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	return client.FetchBalanceByAsset(address, true, client.Asset.GetChain().Chain)
+}
+
+func (client *Client) FetchDecimals(ctx context.Context, contract xc.ContractAddress) (int, error) {
+	return int(client.Asset.GetChain().GetDecimals()), nil
+}
+
+func (client *Client) GetTransactionUrl(txHash string) string {
+	return fmt.Sprintf("%s/transactions/%s", client.Url, txHash)
+}
+
+func (client *Client) GetAccountUrl(address string) string {
+	return fmt.Sprintf("%s/accounts/%s", client.Url, address)
+}
+
+func (client *Client) GetLedgerUrl(sequence uint64) string {
+	return fmt.Sprintf("%s/ledgers/%d", client.Url, sequence)
+}
+
+func (client *Client) GetLatestLedgerUrl() string {
+	return fmt.Sprintf("%s/ledgers?order=desc&limit=1")
+}
+
+// Send a POST request
+func (client *Client) Post(url string, requestBody any, response any) error {
+	jsonPayload, err := json.Marshal(requestBody)
+	if err != nil {
+		return fmt.Errorf("failed to marshal request payload: %w", err)
+	}
+
+	resp, err := client.HttpClient.Post(url, "application/json", bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to post, HTTP status: %s", resp.Status)
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return fmt.Errorf("failed to decode response body: %w", err)
+	}
```

Comment:
Is there a body for error responses?  E.g. often there's a `status` + `message`.  Would be helpful to catch those and include in the error.